torch/fx/experimental/symbolic_shapes.py

```diff
@@ -0,0 +1,90 @@
+import torch
+
+try:
+    import sympy
+    HAS_SYMPY = True
+except ImportError:
+    HAS_SYMPY = False
+
+class PySymInt(object):
+    def __init__(self, expr, shape_env):
+        self.expr = expr
+        self.shape_env = shape_env
+
+    def wrap(self, num):
+        return PySymInt(sympy.Integer(num), self.shape_env)
+
+    def __str__(self):
+        return f"PySymInt({self.expr})"
+
+    def __int__(self):
+        raise RuntimeError("Trying to extract a concrete int out of a symbolic int")
+
+        return self.shape_env.evaluate_expr(self.expr)
+
+    def __bool__(self):
+        return bool(self.shape_env.evaluate_expr(self.expr))
```

Comment:
looks inconsistent 🧐

Reply:
Yeah I'll add a comment. But basically, coercing to bool is generally reasonable. Coercing to int, on the other hand, is a very easy way of shooting yourself in the foot.

Will add a comment.

Later we'll probably need a way to say "I know what I'm doing, please coerce to an int", but that's for later :)